shuyangzhou/liferay-portal
# [Sidecar]LPS-103577

Hi Shuyang,

This pulls includes all the features required by sidecar.

Steps to test:

1. Run ant -f build-test-elasticsearch7.xml start-elasticsearch to build ES manually, will see a folder created in tomcat bundle home folder.
2. Rename the folder to elasticsearch7, and remove osgi/configs/com.liferay.portal.search.elasticsearch7.configuration.ElasticsearchConfiguration.config.
3. Start portal, login and go to control panel->configuration->search, will see Search Engine Vendor: Elasticsearch (Embedded), Client Version: 7.3.0, Nodes: LiferayElasticsearchCluster: [liferay (7.4.1)]

To test cluster,
Just enable cluster by setting "cluster.link.enabled=true"

Known issues:

1. Random failures on ServletContextHelperRegistrationImpl when shutting down, not sure if this is new or just triggered by my change, see https://gist.github.com/tinatian/8ae8776438a065be7fd2c677418ea9c6.
Elasticsearch does not allow to run on root user, needs to bypass
2. Warning logging on startup 
`[2020-01-14T11:22:29,405][WARN ][o.e.d.b.Bootstrap        ] [liferay]future versions of Elasticsearch will require Java 11; your Java version from [/opt/java/jdk-8u222/jre] does not meet this requirement`
3. Dumping log4j.properties into temp folder
4. Clean up the configuration file, remove unused properties
5. Test on different OS.

Thanks.
Tina.
